servers/src/main/java/tachyon/worker/block/evictor/LRUEvictor.java

```diff
@@ -84,7 +86,32 @@ public EvictionPlan freeSpace(long availableBytes, BlockStoreLocation location)
 
     // enough free space
     if (dirCandidates.maxAvailableBytes() >= toEvictBytes) {
+      // candidate blockIds for eviction from current tier
       toEvict = dirCandidates.toEvict();
+      // move as many blocks to next tier as possible
+      int nextTierLevel = mMeta.getBlockMeta(toEvict.get(0)).getBlockLocation().tierLevel() + 1;
+      List<StorageTier> tiers = mMeta.getTiers();
+      int nTotalTiers = tiers.size();
+      while (nextTierLevel < nTotalTiers && toEvict.size() > 0) {
+        // find space in next tier
+        List<StorageDir> dirs = tiers.get(nextTierLevel).getStorageDirs();
+        for (StorageDir dir : dirs) {
+          BlockStoreLocation dest = dir.toBlockStoreLocation();
+          Iterator<Long> blocks = toEvict.iterator();
```

Comment:
Is this sorted in any order? If possible, it would be more correct to spill the most recently used blocks to the next tier first.